client/client.go

```diff
@@ -517,21 +518,21 @@ func (c *authenticatingClient) doAuthenticate() error {
 		return nil
 	}
 	if c.authMode == nil {
-		return fmt.Errorf("Authentication method has not been specified")
+		return gooseerrors.Newf(nil, "Authentication method has not been specified")
 	}
 	var (
 		authDetails *identity.AuthDetails
 		err         error
 	)
 	if authDetails, err = c.authMode.Auth(c.creds); err != nil {
-		return gooseerrors.Newf(err, "authentication failed")
+		return gooseerrors.NewUnauthorisedf(err, "", err.Error())
```

Comment:
This is the wrong place - take a look at the underlying code. There can be an error making the json request which is not an auth error but a connectivity error. The place where the actual auth error is determined is the place to use gooseerrors.NewUnauthorisedf